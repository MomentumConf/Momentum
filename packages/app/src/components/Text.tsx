import styled from '@emotion/native'

const CustomText = styled.Text`
  font-size: ${({ theme }) => `${theme.fontSize.m}px`};
  color: ${({ theme }) => theme.color.text};
  font-family: ${({ theme }) => theme.font.default};
`

export default CustomText
