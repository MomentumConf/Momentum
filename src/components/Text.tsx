import styled from '@emotion/native'
import React, { ComponentPropsWithoutRef, FC } from 'react'
import { Text } from 'react-native'

const fontFamily = {
  regular: 'Proxima',
  light: 'Proxima-Light',
  bold: 'Proxima-Black',
}

type TextProps = {
  fontWeight?: keyof typeof fontFamily
}

const StyledText = styled.Text<TextProps>`
  color: ${({ theme }) => theme.colors.text};
  font-family: ${({ fontWeight }) => fontFamily[fontWeight || 'regular']};
`

type CustomTextProps = ComponentPropsWithoutRef<typeof Text> & Partial<TextProps>

const CustomText: FC<CustomTextProps> = ({ fontWeight = 'regular', ...props }) => (
  <StyledText {...props} fontWeight={fontWeight} />
)

export default CustomText
