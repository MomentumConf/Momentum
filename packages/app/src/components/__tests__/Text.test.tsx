import React from 'react'
import renderer from 'utils/testRenderer'

import Text from '../Text'

describe('component StyledText', () => {
  it(`renders correctly`, () => {
    expect.assertions(1)
    const tree = renderer(<Text>Snapshot test!</Text>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
