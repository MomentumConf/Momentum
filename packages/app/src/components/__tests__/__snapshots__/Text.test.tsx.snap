// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component StyledText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#121212",
        "fontFamily": "Proxima",
        "fontSize": 18,
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
